src/FindTranslocations.cpp

```diff
@@ -15,165 +15,47 @@
-				string regionfile = "";
-
-				try{
-					//extracts the startpos,endpos and chromosome of each given region
-					for(int i=0;i<=1;i++){
-						//extracts the chromosome REFID
-						tmpstr=splitline.front();
-						splitline.erase( splitline.begin() );
-						chr.push(contig2position[tmpstr]);
-						regionfile+=tmpstr;
-						regionfile+="_";
-			
-						//extracts the start position on the given chromosome
-						tmpstr=splitline.front();
-						splitline.erase( splitline.begin() );
-						startPos.push( atoi(tmpstr.c_str()));
-						regionfile+=tmpstr;
-						regionfile+="_";
-
-						//extracts the end position on the given chromosome
-						tmpstr=splitline.front();
-						splitline.erase( splitline.begin() );
-						endPos.push(atoi(tmpstr.c_str()));
-						regionfile+=tmpstr;
-						if(i == 0){regionfile+="_";}
-
-
-						if(startPos.back() > endPos.back()){
-							cout << "ERROR: chromosome startpos must be located before endpos" << endl;
-							return;
-						}
-			
-					}
-				}catch(...){
-					cout << "error invalid --extract input" << endl;
-					return;
-				}
-
-				regionfile+=".bam";
-				regionfile=eventfolder+"/"+regionfile;
-				cout << regionfile << endl;
-					// attempt to open our BamWriter{
-					if ( !regionwriter.Open(regionfile.c_str(), header, references) ) {
-						cout << "Could not open output event BAM file" << endl;
-						return;
-					}
-				
-				//test if an index file is available
-				if(bamFile.LocateIndex() == 0){
-					cout << "warning no index file found, extraction will proceed in slow mode" << endl;
-				}
-
-
-
-
-				//collects all reads found in the given intervals and writes them to the file outputFileHeader+"_roi.bam"
-				for(int i=0;i<n;i++){
-					bamFile.SetRegion(chr.front(),startPos.front(),chr.front(),endPos.front()); 
-					chr.pop();startPos.pop();endPos.pop();
-					while ( bamFile.GetNextAlignment(currentRead) ) {
-						if(currentRead.IsMapped()) {
-						//prints the read to bam file
-						writer.SaveAlignment(currentRead);
-						regionwriter.SaveAlignment(currentRead);
-						}
-					}
-				}
-				regionwriter.Close();
-				
-
-			}
-			j++;
-		}
-		
-		inputFile.close();
+	files.push(vm["bam"].as<string>());
+	//alignmentFile = vm["bam"].as<string>();
+	if (!vm.count("bai")){
+		files.push("error");
+		return(files);
+	}
+	//string indexFile=vm["bai"].as<string>();
+	files.push(vm["bai"].as<string>());
+	if (vm.count("output")) {
+		//string header = vm["output"].as<string>();
+		//outputFileHeader = header ;
+		files.push(vm["output"].as<string>());
+	}else{
+		files.push("output"); // default output name
 	}
 
-	cout << "writing complete" << endl;
-	cout << outputFileHeader << endl;
-	writer.Close();
-	bamFile.Close();
-	
-
+	return(files);
 }
 
+//Function used to extract reads in specified regions and write the reads to bam files
+void extract(string BamFileName,string outputFileHeader,string inputFileName,map<string,unsigned int> contig2position, string indexFile);
```

Comment:
Try to be more specific with function names, in two months from now you will look at the function `extract` called in the middle of the code and have no clue about what the function does :-D